From 711f9256517966629ed3cb6e36fe45a1c014b3e7 Mon Sep 17 00:00:00 2001
From: yuan junhui <yuanjunhui@eswincomputing.com>
Date: Wed, 15 May 2024 14:23:12 +0800
Subject: [PATCH 032/109] WIN2030-15279:fix(bmc):fixbug Unit,del comments

Changelogs:
01,temperature,RPM
02,delete Chinese comments
03,add login,modify css

Change-Id: I0eb65038d91b04e1182af45ebfcd5528d158e21b
---
 Core/Src/web-server.c | 760 +++++-------------------------------------
 1 file changed, 88 insertions(+), 672 deletions(-)

diff --git a/Core/Src/web-server.c b/Core/Src/web-server.c
index b52e219..e522b9f 100644
--- a/Core/Src/web-server.c
+++ b/Core/Src/web-server.c
@@ -16,12 +16,11 @@
 #define BUF_SIZE_128 128
 #define BUF_SIZE_256 256
 #define BUF_SIZE_512 512
-#define MAX_AGE 60*3
+#define MAX_AGE 60*3 //auto logout timeout
 
-#define STATIC_PATH "/tmp/static"
 #define EEPROM_USERNAME_PASSWORD_ADDR 0x0100
 #define EEPROM_USERNAME_PASSWORD_BUFFER_SIZE 64
-#define AT24C_ADDR (0x50<<1) //todoooooooooooooooooo
+
 #define CHUNK_SIZE 1024
 #define MAX_YEAR 3000
 
@@ -8981,450 +8980,6 @@ static const unsigned char data_jquery_min_js[] = {
 	0x65, 0x72, 0x79, 0x3D, 0x43, 0x2E, 0x24, 0x3D, 0x53, 0x29,
 	0x2C, 0x53, 0x7D, 0x29, 0x3B, 0x0A};
 
-static const unsigned char data_404_html[] = {
-	/* File: /404.html */
-	0x2F, 0x34, 0x30, 0x34, 0x2E, 0x68, 0x74, 0x6D, 0x6C, 0,
-	/* here starts the data */
-	0x3C, 0x21, 0x44, 0x4F, 0x43, 0x54, 0x59, 0x50, 0x45, 0x20,
-	0x68, 0x74, 0x6D, 0x6C, 0x3E, 0x0D, 0x0A, 0x3C, 0x68, 0x74,
-	0x6D, 0x6C, 0x20, 0x6C, 0x61, 0x6E, 0x67, 0x3D, 0x22, 0x65,
-	0x6E, 0x22, 0x3E, 0x0D, 0x0A, 0x0D, 0x0A, 0x3C, 0x68, 0x65,
-	0x61, 0x64, 0x3E, 0x0D, 0x0A, 0x20, 0x20, 0x3C, 0x6D, 0x65,
-	0x74, 0x61, 0x20, 0x63, 0x68, 0x61, 0x72, 0x73, 0x65, 0x74,
-	0x3D, 0x22, 0x55, 0x54, 0x46, 0x2D, 0x38, 0x22, 0x3E, 0x0D,
-	0x0A, 0x20, 0x20, 0x3C, 0x6D, 0x65, 0x74, 0x61, 0x20, 0x68,
-	0x74, 0x74, 0x70, 0x2D, 0x65, 0x71, 0x75, 0x69, 0x76, 0x3D,
-	0x22, 0x58, 0x2D, 0x55, 0x41, 0x2D, 0x43, 0x6F, 0x6D, 0x70,
-	0x61, 0x74, 0x69, 0x62, 0x6C, 0x65, 0x22, 0x20, 0x63, 0x6F,
-	0x6E, 0x74, 0x65, 0x6E, 0x74, 0x3D, 0x22, 0x49, 0x45, 0x3D,
-	0x65, 0x64, 0x67, 0x65, 0x22, 0x3E, 0x0D, 0x0A, 0x20, 0x20,
-	0x3C, 0x6D, 0x65, 0x74, 0x61, 0x20, 0x6E, 0x61, 0x6D, 0x65,
-	0x3D, 0x22, 0x76, 0x69, 0x65, 0x77, 0x70, 0x6F, 0x72, 0x74,
-	0x22, 0x20, 0x63, 0x6F, 0x6E, 0x74, 0x65, 0x6E, 0x74, 0x3D,
-	0x22, 0x77, 0x69, 0x64, 0x74, 0x68, 0x3D, 0x64, 0x65, 0x76,
-	0x69, 0x63, 0x65, 0x2D, 0x77, 0x69, 0x64, 0x74, 0x68, 0x2C,
-	0x20, 0x69, 0x6E, 0x69, 0x74, 0x69, 0x61, 0x6C, 0x2D, 0x73,
-	0x63, 0x61, 0x6C, 0x65, 0x3D, 0x31, 0x2E, 0x30, 0x22, 0x3E,
-	0x0D, 0x0A, 0x20, 0x20, 0x3C, 0x74, 0x69, 0x74, 0x6C, 0x65,
-	0x3E, 0x45, 0x72, 0x72, 0x6F, 0x72, 0x20, 0x34, 0x30, 0x34,
-	0x20, 0x70, 0x61, 0x67, 0x65, 0x3C, 0x2F, 0x74, 0x69, 0x74,
-	0x6C, 0x65, 0x3E, 0x0D, 0x0A, 0x20, 0x20, 0x3C, 0x73, 0x74,
-	0x79, 0x6C, 0x65, 0x3E, 0x0D, 0x0A, 0x20, 0x20, 0x20, 0x20,
-	0x62, 0x6F, 0x64, 0x79, 0x20, 0x7B, 0x0D, 0x0A, 0x20, 0x20,
-	0x20, 0x20, 0x20, 0x20, 0x6D, 0x61, 0x72, 0x67, 0x69, 0x6E,
-	0x3A, 0x20, 0x30, 0x3B, 0x0D, 0x0A, 0x20, 0x20, 0x20, 0x20,
-	0x20, 0x20, 0x62, 0x6F, 0x78, 0x2D, 0x73, 0x69, 0x7A, 0x69,
-	0x6E, 0x67, 0x3A, 0x20, 0x62, 0x6F, 0x72, 0x64, 0x65, 0x72,
-	0x2D, 0x62, 0x6F, 0x78, 0x3B, 0x0D, 0x0A, 0x20, 0x20, 0x20,
-	0x20, 0x7D, 0x0D, 0x0A, 0x0D, 0x0A, 0x20, 0x20, 0x20, 0x20,
-	0x2E, 0x63, 0x6F, 0x6E, 0x74, 0x61, 0x69, 0x6E, 0x65, 0x72,
-	0x20, 0x7B, 0x0D, 0x0A, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20,
-	0x68, 0x65, 0x69, 0x67, 0x68, 0x74, 0x3A, 0x20, 0x38, 0x30,
-	0x76, 0x68, 0x3B, 0x0D, 0x0A, 0x20, 0x20, 0x20, 0x20, 0x20,
-	0x20, 0x64, 0x69, 0x73, 0x70, 0x6C, 0x61, 0x79, 0x3A, 0x20,
-	0x66, 0x6C, 0x65, 0x78, 0x3B, 0x0D, 0x0A, 0x20, 0x20, 0x20,
-	0x20, 0x20, 0x20, 0x6A, 0x75, 0x73, 0x74, 0x69, 0x66, 0x79,
-	0x2D, 0x63, 0x6F, 0x6E, 0x74, 0x65, 0x6E, 0x74, 0x3A, 0x20,
-	0x63, 0x65, 0x6E, 0x74, 0x65, 0x72, 0x3B, 0x0D, 0x0A, 0x20,
-	0x20, 0x20, 0x20, 0x20, 0x20, 0x61, 0x6C, 0x69, 0x67, 0x6E,
-	0x2D, 0x69, 0x74, 0x65, 0x6D, 0x73, 0x3A, 0x20, 0x63, 0x65,
-	0x6E, 0x74, 0x65, 0x72, 0x3B, 0x0D, 0x0A, 0x20, 0x20, 0x20,
-	0x20, 0x20, 0x20, 0x66, 0x6C, 0x65, 0x78, 0x2D, 0x64, 0x69,
-	0x72, 0x65, 0x63, 0x74, 0x69, 0x6F, 0x6E, 0x3A, 0x20, 0x63,
-	0x6F, 0x6C, 0x75, 0x6D, 0x6E, 0x3B, 0x0D, 0x0A, 0x20, 0x20,
-	0x20, 0x20, 0x20, 0x20, 0x62, 0x61, 0x63, 0x6B, 0x67, 0x72,
-	0x6F, 0x75, 0x6E, 0x64, 0x3A, 0x20, 0x6C, 0x69, 0x6E, 0x65,
-	0x61, 0x72, 0x2D, 0x67, 0x72, 0x61, 0x64, 0x69, 0x65, 0x6E,
-	0x74, 0x28, 0x74, 0x6F, 0x20, 0x72, 0x69, 0x67, 0x68, 0x74,
-	0x2C, 0x20, 0x72, 0x67, 0x62, 0x61, 0x28, 0x30, 0x2C, 0x20,
-	0x30, 0x2C, 0x20, 0x30, 0x2C, 0x20, 0x2E, 0x35, 0x29, 0x2C,
-	0x20, 0x72, 0x67, 0x62, 0x61, 0x28, 0x30, 0x2C, 0x20, 0x30,
-	0x2C, 0x20, 0x30, 0x2C, 0x20, 0x2E, 0x31, 0x29, 0x29, 0x2C,
-	0x20, 0x75, 0x72, 0x6C, 0x28, 0x27, 0x68, 0x74, 0x74, 0x70,
-	0x73, 0x3A, 0x2F, 0x2F, 0x69, 0x6D, 0x61, 0x67, 0x65, 0x73,
-	0x2E, 0x75, 0x6E, 0x73, 0x70, 0x6C, 0x61, 0x73, 0x68, 0x2E,
-	0x63, 0x6F, 0x6D, 0x2F, 0x70, 0x68, 0x6F, 0x74, 0x6F, 0x2D,
-	0x31, 0x35, 0x39, 0x35, 0x36, 0x32, 0x34, 0x38, 0x37, 0x31,
-	0x39, 0x33, 0x30, 0x2D, 0x36, 0x65, 0x38, 0x35, 0x33, 0x37,
-	0x39, 0x39, 0x38, 0x35, 0x39, 0x32, 0x3F, 0x69, 0x78, 0x6C,
-	0x69, 0x62, 0x3D, 0x72, 0x62, 0x2D, 0x31, 0x2E, 0x32, 0x2E,
-	0x31, 0x26, 0x69, 0x78, 0x69, 0x64, 0x3D, 0x4D, 0x6E, 0x77,
-	0x78, 0x4D, 0x6A, 0x41, 0x33, 0x66, 0x44, 0x42, 0x38, 0x4D,
-	0x48, 0x78, 0x77, 0x61, 0x47, 0x39, 0x30, 0x62, 0x79, 0x31,
-	0x77, 0x59, 0x57, 0x64, 0x6C, 0x66, 0x48, 0x78, 0x38, 0x66,
-	0x47, 0x56, 0x75, 0x66, 0x44, 0x42, 0x38, 0x66, 0x48, 0x78,
-	0x38, 0x26, 0x61, 0x75, 0x74, 0x6F, 0x3D, 0x66, 0x6F, 0x72,
-	0x6D, 0x61, 0x74, 0x26, 0x66, 0x69, 0x74, 0x3D, 0x63, 0x72,
-	0x6F, 0x70, 0x26, 0x77, 0x3D, 0x38, 0x37, 0x31, 0x26, 0x71,
-	0x3D, 0x38, 0x30, 0x27, 0x29, 0x3B, 0x0D, 0x0A, 0x20, 0x20,
-	0x20, 0x20, 0x20, 0x20, 0x62, 0x61, 0x63, 0x6B, 0x67, 0x72,
-	0x6F, 0x75, 0x6E, 0x64, 0x2D, 0x73, 0x69, 0x7A, 0x65, 0x3A,
-	0x20, 0x63, 0x6F, 0x76, 0x65, 0x72, 0x3B, 0x0D, 0x0A, 0x20,
-	0x20, 0x20, 0x20, 0x20, 0x20, 0x62, 0x61, 0x63, 0x6B, 0x67,
-	0x72, 0x6F, 0x75, 0x6E, 0x64, 0x2D, 0x70, 0x6F, 0x73, 0x69,
-	0x74, 0x69, 0x6F, 0x6E, 0x3A, 0x20, 0x63, 0x65, 0x6E, 0x74,
-	0x65, 0x72, 0x3B, 0x0D, 0x0A, 0x20, 0x20, 0x20, 0x20, 0x20,
-	0x20, 0x62, 0x61, 0x63, 0x6B, 0x67, 0x72, 0x6F, 0x75, 0x6E,
-	0x64, 0x2D, 0x72, 0x65, 0x70, 0x65, 0x61, 0x74, 0x3A, 0x20,
-	0x6E, 0x6F, 0x2D, 0x72, 0x65, 0x70, 0x65, 0x61, 0x74, 0x3B,
-	0x0D, 0x0A, 0x20, 0x20, 0x20, 0x20, 0x7D, 0x0D, 0x0A, 0x0D,
-	0x0A, 0x20, 0x20, 0x20, 0x20, 0x68, 0x31, 0x20, 0x7B, 0x0D,
-	0x0A, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x66, 0x6F, 0x6E,
-	0x74, 0x2D, 0x73, 0x69, 0x7A, 0x65, 0x3A, 0x20, 0x31, 0x30,
-	0x72, 0x65, 0x6D, 0x3B, 0x0D, 0x0A, 0x20, 0x20, 0x20, 0x20,
-	0x20, 0x20, 0x63, 0x6F, 0x6C, 0x6F, 0x72, 0x3A, 0x20, 0x23,
-	0x66, 0x66, 0x66, 0x3B, 0x0D, 0x0A, 0x20, 0x20, 0x20, 0x20,
-	0x20, 0x20, 0x74, 0x65, 0x78, 0x74, 0x2D, 0x74, 0x72, 0x61,
-	0x6E, 0x73, 0x66, 0x6F, 0x72, 0x6D, 0x3A, 0x20, 0x75, 0x70,
-	0x70, 0x65, 0x72, 0x63, 0x61, 0x73, 0x65, 0x3B, 0x0D, 0x0A,
-	0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x66, 0x6F, 0x6E, 0x74,
-	0x2D, 0x77, 0x65, 0x69, 0x67, 0x68, 0x74, 0x3A, 0x20, 0x37,
-	0x30, 0x30, 0x3B, 0x0D, 0x0A, 0x20, 0x20, 0x20, 0x20, 0x20,
-	0x20, 0x6D, 0x61, 0x72, 0x67, 0x69, 0x6E, 0x2D, 0x62, 0x6F,
-	0x74, 0x74, 0x6F, 0x6D, 0x3A, 0x20, 0x31, 0x72, 0x65, 0x6D,
-	0x3B, 0x0D, 0x0A, 0x20, 0x20, 0x20, 0x20, 0x7D, 0x0D, 0x0A,
-	0x0D, 0x0A, 0x20, 0x20, 0x20, 0x20, 0x68, 0x32, 0x20, 0x7B,
-	0x0D, 0x0A, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x66, 0x6F,
-	0x6E, 0x74, 0x2D, 0x73, 0x69, 0x7A, 0x65, 0x3A, 0x20, 0x32,
-	0x72, 0x65, 0x6D, 0x3B, 0x0D, 0x0A, 0x20, 0x20, 0x20, 0x20,
-	0x20, 0x20, 0x63, 0x6F, 0x6C, 0x6F, 0x72, 0x3A, 0x20, 0x23,
-	0x66, 0x66, 0x66, 0x3B, 0x0D, 0x0A, 0x20, 0x20, 0x20, 0x20,
-	0x20, 0x20, 0x74, 0x65, 0x78, 0x74, 0x2D, 0x74, 0x72, 0x61,
-	0x6E, 0x73, 0x66, 0x6F, 0x72, 0x6D, 0x3A, 0x20, 0x75, 0x70,
-	0x70, 0x65, 0x72, 0x63, 0x61, 0x73, 0x65, 0x3B, 0x0D, 0x0A,
-	0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x66, 0x6F, 0x6E, 0x74,
-	0x2D, 0x77, 0x65, 0x69, 0x67, 0x68, 0x74, 0x3A, 0x20, 0x37,
-	0x30, 0x30, 0x3B, 0x0D, 0x0A, 0x20, 0x20, 0x20, 0x20, 0x20,
-	0x20, 0x6D, 0x61, 0x72, 0x67, 0x69, 0x6E, 0x2D, 0x62, 0x6F,
-	0x74, 0x74, 0x6F, 0x6D, 0x3A, 0x20, 0x31, 0x72, 0x65, 0x6D,
-	0x3B, 0x0D, 0x0A, 0x20, 0x20, 0x20, 0x20, 0x7D, 0x0D, 0x0A,
-	0x0D, 0x0A, 0x20, 0x20, 0x20, 0x20, 0x70, 0x20, 0x7B, 0x0D,
-	0x0A, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x66, 0x6F, 0x6E,
-	0x74, 0x2D, 0x73, 0x69, 0x7A, 0x65, 0x3A, 0x20, 0x31, 0x2E,
-	0x35, 0x72, 0x65, 0x6D, 0x3B, 0x0D, 0x0A, 0x20, 0x20, 0x20,
-	0x20, 0x20, 0x20, 0x63, 0x6F, 0x6C, 0x6F, 0x72, 0x3A, 0x20,
-	0x23, 0x66, 0x66, 0x66, 0x3B, 0x0D, 0x0A, 0x20, 0x20, 0x20,
-	0x20, 0x20, 0x20, 0x66, 0x6F, 0x6E, 0x74, 0x2D, 0x77, 0x65,
-	0x69, 0x67, 0x68, 0x74, 0x3A, 0x20, 0x37, 0x30, 0x30, 0x3B,
-	0x0D, 0x0A, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x6D, 0x61,
-	0x72, 0x67, 0x69, 0x6E, 0x2D, 0x62, 0x6F, 0x74, 0x74, 0x6F,
-	0x6D, 0x3A, 0x20, 0x31, 0x72, 0x65, 0x6D, 0x3B, 0x0D, 0x0A,
-	0x20, 0x20, 0x20, 0x20, 0x7D, 0x0D, 0x0A, 0x0D, 0x0A, 0x20,
-	0x20, 0x20, 0x20, 0x61, 0x20, 0x7B, 0x0D, 0x0A, 0x20, 0x20,
-	0x20, 0x20, 0x20, 0x20, 0x70, 0x61, 0x64, 0x64, 0x69, 0x6E,
-	0x67, 0x3A, 0x20, 0x31, 0x35, 0x70, 0x78, 0x20, 0x32, 0x30,
-	0x70, 0x78, 0x3B, 0x0D, 0x0A, 0x20, 0x20, 0x20, 0x20, 0x20,
-	0x20, 0x62, 0x61, 0x63, 0x6B, 0x67, 0x72, 0x6F, 0x75, 0x6E,
-	0x64, 0x3A, 0x20, 0x23, 0x35, 0x32, 0x63, 0x61, 0x65, 0x65,
-	0x3B, 0x0D, 0x0A, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x66,
-	0x6F, 0x6E, 0x74, 0x2D, 0x73, 0x69, 0x7A, 0x65, 0x3A, 0x20,
-	0x31, 0x72, 0x65, 0x6D, 0x3B, 0x0D, 0x0A, 0x20, 0x20, 0x20,
-	0x20, 0x20, 0x20, 0x74, 0x65, 0x78, 0x74, 0x2D, 0x64, 0x65,
-	0x63, 0x6F, 0x72, 0x61, 0x74, 0x69, 0x6F, 0x6E, 0x3A, 0x20,
-	0x6E, 0x6F, 0x6E, 0x65, 0x3B, 0x0D, 0x0A, 0x20, 0x20, 0x20,
-	0x20, 0x20, 0x20, 0x63, 0x6F, 0x6C, 0x6F, 0x72, 0x3A, 0x20,
-	0x23, 0x33, 0x33, 0x33, 0x33, 0x33, 0x33, 0x3B, 0x0D, 0x0A,
-	0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x62, 0x6F, 0x72, 0x64,
-	0x65, 0x72, 0x2D, 0x72, 0x61, 0x64, 0x69, 0x75, 0x73, 0x3A,
-	0x20, 0x2E, 0x32, 0x35, 0x72, 0x65, 0x6D, 0x3B, 0x0D, 0x0A,
-	0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x62, 0x6F, 0x78, 0x2D,
-	0x73, 0x68, 0x61, 0x64, 0x6F, 0x77, 0x3A, 0x20, 0x30, 0x20,
-	0x30, 0x20, 0x32, 0x30, 0x70, 0x78, 0x20, 0x72, 0x67, 0x62,
-	0x61, 0x28, 0x32, 0x35, 0x35, 0x2C, 0x20, 0x32, 0x35, 0x35,
-	0x2C, 0x20, 0x32, 0x35, 0x35, 0x2C, 0x20, 0x30, 0x2E, 0x38,
-	0x30, 0x38, 0x29, 0x0D, 0x0A, 0x20, 0x20, 0x20, 0x20, 0x7D,
-	0x0D, 0x0A, 0x20, 0x20, 0x3C, 0x2F, 0x73, 0x74, 0x79, 0x6C,
-	0x65, 0x3E, 0x0D, 0x0A, 0x3C, 0x2F, 0x68, 0x65, 0x61, 0x64,
-	0x3E, 0x0D, 0x0A, 0x0D, 0x0A, 0x3C, 0x62, 0x6F, 0x64, 0x79,
-	0x3E, 0x0D, 0x0A, 0x20, 0x20, 0x3C, 0x64, 0x69, 0x76, 0x20,
-	0x63, 0x6C, 0x61, 0x73, 0x73, 0x3D, 0x22, 0x63, 0x6F, 0x6E,
-	0x74, 0x61, 0x69, 0x6E, 0x65, 0x72, 0x22, 0x3E, 0x0D, 0x0A,
-	0x20, 0x20, 0x20, 0x20, 0x3C, 0x68, 0x31, 0x3E, 0x34, 0x30,
-	0x34, 0x3C, 0x2F, 0x68, 0x31, 0x3E, 0x0D, 0x0A, 0x20, 0x20,
-	0x20, 0x20, 0x3C, 0x68, 0x32, 0x3E, 0x50, 0x61, 0x67, 0x65,
-	0x20, 0x6E, 0x6F, 0x74, 0x20, 0x66, 0x6F, 0x75, 0x6E, 0x64,
-	0x3C, 0x2F, 0x68, 0x32, 0x3E, 0x0D, 0x0A, 0x20, 0x20, 0x20,
-	0x20, 0x3C, 0x70, 0x3E, 0x54, 0x68, 0x65, 0x20, 0x70, 0x61,
-	0x67, 0x65, 0x20, 0x79, 0x6F, 0x75, 0x20, 0x61, 0x72, 0x65,
-	0x20, 0x6C, 0x6F, 0x6F, 0x6B, 0x69, 0x6E, 0x67, 0x20, 0x66,
-	0x6F, 0x72, 0x20, 0x64, 0x6F, 0x65, 0x73, 0x20, 0x6E, 0x6F,
-	0x74, 0x20, 0x65, 0x78, 0x69, 0x73, 0x74, 0x2E, 0x3C, 0x2F,
-	0x70, 0x3E, 0x0D, 0x0A, 0x20, 0x20, 0x20, 0x20, 0x3C, 0x70,
-	0x3E, 0x3C, 0x61, 0x20, 0x68, 0x72, 0x65, 0x66, 0x3D, 0x22,
-	0x68, 0x74, 0x74, 0x70, 0x73, 0x3A, 0x2F, 0x2F, 0x77, 0x77,
-	0x77, 0x2E, 0x65, 0x6D, 0x62, 0x65, 0x74, 0x72, 0x6F, 0x6E,
-	0x69, 0x63, 0x78, 0x2E, 0x63, 0x6F, 0x6D, 0x2F, 0x22, 0x3E,
-	0x56, 0x69, 0x73, 0x69, 0x74, 0x20, 0x45, 0x6D, 0x62, 0x65,
-	0x54, 0x72, 0x6F, 0x6E, 0x69, 0x63, 0x58, 0x3C, 0x2F, 0x61,
-	0x3E, 0x3C, 0x2F, 0x70, 0x3E, 0x0D, 0x0A, 0x20, 0x20, 0x3C,
-	0x2F, 0x64, 0x69, 0x76, 0x3E, 0x0D, 0x0A, 0x3C, 0x2F, 0x62,
-	0x6F, 0x64, 0x79, 0x3E, 0x0D, 0x0A, 0x0D, 0x0A, 0x3C, 0x2F,
-	0x68, 0x74, 0x6D, 0x6C, 0x3E};
-
-static const unsigned char data_index_html[] = {
-	/* File: /index.html */
-	0x2F, 0x69, 0x6E, 0x64, 0x65, 0x78, 0x2E, 0x68, 0x74, 0x6D, 0x6C, 0,
-	/* here starts the data */
-	0x3C, 0x68, 0x74, 0x6D, 0x6C, 0x3E, 0x0D, 0x0A, 0x3C, 0x68,
-	0x65, 0x61, 0x64, 0x3E, 0x0D, 0x0A, 0x20, 0x20, 0x20, 0x3C,
-	0x73, 0x74, 0x79, 0x6C, 0x65, 0x3E, 0x0D, 0x0A, 0x20, 0x20,
-	0x20, 0x20, 0x20, 0x20, 0x20, 0x62, 0x6F, 0x64, 0x79, 0x7B,
-	0x0D, 0x0A, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20,
-	0x66, 0x6F, 0x6E, 0x74, 0x2D, 0x73, 0x69, 0x7A, 0x65, 0x3A,
-	0x32, 0x35, 0x70, 0x78, 0x3B, 0x0D, 0x0A, 0x20, 0x20, 0x20,
-	0x20, 0x20, 0x20, 0x20, 0x20, 0x63, 0x6F, 0x6C, 0x6F, 0x72,
-	0x3A, 0x20, 0x23, 0x30, 0x36, 0x35, 0x35, 0x33, 0x35, 0x3B,
-	0x0D, 0x0A, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20,
-	0x62, 0x61, 0x63, 0x6B, 0x67, 0x72, 0x6F, 0x75, 0x6E, 0x64,
-	0x2D, 0x73, 0x69, 0x7A, 0x65, 0x3A, 0x20, 0x63, 0x6F, 0x76,
-	0x65, 0x72, 0x3B, 0x0D, 0x0A, 0x20, 0x20, 0x20, 0x20, 0x20,
-	0x20, 0x20, 0x7D, 0x0D, 0x0A, 0x20, 0x20, 0x20, 0x20, 0x20,
-	0x20, 0x2E, 0x77, 0x64, 0x7B, 0x0D, 0x0A, 0x20, 0x20, 0x20,
-	0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x68, 0x65, 0x69,
-	0x67, 0x68, 0x74, 0x3A, 0x20, 0x31, 0x30, 0x30, 0x76, 0x68,
-	0x3B, 0x0D, 0x0A, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20,
-	0x20, 0x20, 0x20, 0x62, 0x61, 0x63, 0x6B, 0x67, 0x72, 0x6F,
-	0x75, 0x6E, 0x64, 0x3A, 0x20, 0x6C, 0x69, 0x6E, 0x65, 0x61,
-	0x72, 0x2D, 0x67, 0x72, 0x61, 0x64, 0x69, 0x65, 0x6E, 0x74,
-	0x28, 0x74, 0x6F, 0x20, 0x72, 0x69, 0x67, 0x68, 0x74, 0x2C,
-	0x20, 0x72, 0x67, 0x62, 0x61, 0x28, 0x30, 0x2C, 0x20, 0x30,
-	0x2C, 0x20, 0x30, 0x2C, 0x20, 0x2E, 0x35, 0x29, 0x2C, 0x20,
-	0x72, 0x67, 0x62, 0x61, 0x28, 0x30, 0x2C, 0x20, 0x30, 0x2C,
-	0x20, 0x30, 0x2C, 0x20, 0x2E, 0x31, 0x29, 0x29, 0x2C, 0x20,
-	0x75, 0x72, 0x6C, 0x28, 0x27, 0x68, 0x74, 0x74, 0x70, 0x3A,
-	0x2F, 0x2F, 0x65, 0x6D, 0x62, 0x65, 0x74, 0x72, 0x6F, 0x6E,
-	0x69, 0x63, 0x78, 0x2E, 0x63, 0x6F, 0x6D, 0x2F, 0x77, 0x70,
-	0x2D, 0x63, 0x6F, 0x6E, 0x74, 0x65, 0x6E, 0x74, 0x2F, 0x75,
-	0x70, 0x6C, 0x6F, 0x61, 0x64, 0x73, 0x2F, 0x32, 0x30, 0x32,
-	0x31, 0x2F, 0x31, 0x30, 0x2F, 0x63, 0x72, 0x6F, 0x70, 0x70,
-	0x65, 0x64, 0x2D, 0x4C, 0x6F, 0x67, 0x6F, 0x2D, 0x74, 0x72,
-	0x61, 0x6E, 0x73, 0x2E, 0x70, 0x6E, 0x67, 0x27, 0x29, 0x3B,
-	0x0D, 0x0A, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20,
-	0x20, 0x20, 0x62, 0x61, 0x63, 0x6B, 0x67, 0x72, 0x6F, 0x75,
-	0x6E, 0x64, 0x2D, 0x72, 0x65, 0x70, 0x65, 0x61, 0x74, 0x3A,
-	0x20, 0x6E, 0x6F, 0x2D, 0x72, 0x65, 0x70, 0x65, 0x61, 0x74,
-	0x3B, 0x0D, 0x0A, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20,
-	0x20, 0x20, 0x20, 0x70, 0x61, 0x64, 0x64, 0x69, 0x6E, 0x67,
-	0x3A, 0x20, 0x31, 0x30, 0x30, 0x70, 0x78, 0x3B, 0x0D, 0x0A,
-	0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x7D, 0x0D, 0x0A, 0x20,
-	0x20, 0x20, 0x20, 0x20, 0x20, 0x2E, 0x77, 0x64, 0x20, 0x68,
-	0x31, 0x7B, 0x0D, 0x0A, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20,
-	0x20, 0x20, 0x20, 0x20, 0x74, 0x65, 0x78, 0x74, 0x2D, 0x61,
-	0x6C, 0x69, 0x67, 0x6E, 0x3A, 0x20, 0x63, 0x65, 0x6E, 0x74,
-	0x65, 0x72, 0x3B, 0x0D, 0x0A, 0x20, 0x20, 0x20, 0x20, 0x20,
-	0x20, 0x20, 0x20, 0x20, 0x20, 0x66, 0x6F, 0x6E, 0x74, 0x2D,
-	0x77, 0x65, 0x69, 0x67, 0x68, 0x74, 0x3A, 0x20, 0x33, 0x30,
-	0x30, 0x70, 0x78, 0x3B, 0x0D, 0x0A, 0x20, 0x20, 0x20, 0x20,
-	0x20, 0x20, 0x7D, 0x0D, 0x0A, 0x20, 0x20, 0x20, 0x20, 0x20,
-	0x20, 0x2E, 0x77, 0x64, 0x20, 0x68, 0x34, 0x7B, 0x0D, 0x0A,
-	0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20,
-	0x74, 0x65, 0x78, 0x74, 0x2D, 0x61, 0x6C, 0x69, 0x67, 0x6E,
-	0x3A, 0x20, 0x63, 0x65, 0x6E, 0x74, 0x65, 0x72, 0x3B, 0x0D,
-	0x0A, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20,
-	0x20, 0x63, 0x6F, 0x6C, 0x6F, 0x72, 0x3A, 0x23, 0x39, 0x39,
-	0x30, 0x30, 0x30, 0x30, 0x3B, 0x0D, 0x0A, 0x20, 0x20, 0x20,
-	0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x66, 0x6F, 0x6E,
-	0x74, 0x2D, 0x77, 0x65, 0x69, 0x67, 0x68, 0x74, 0x3A, 0x20,
-	0x31, 0x30, 0x30, 0x70, 0x78, 0x3B, 0x0D, 0x0A, 0x20, 0x20,
-	0x20, 0x20, 0x20, 0x20, 0x7D, 0x0D, 0x0A, 0x20, 0x20, 0x20,
-	0x20, 0x20, 0x20, 0x2E, 0x77, 0x64, 0x20, 0x68, 0x35, 0x7B,
-	0x0D, 0x0A, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20,
-	0x74, 0x65, 0x78, 0x74, 0x2D, 0x61, 0x6C, 0x69, 0x67, 0x6E,
-	0x3A, 0x20, 0x63, 0x65, 0x6E, 0x74, 0x65, 0x72, 0x3B, 0x0D,
-	0x0A, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x66,
-	0x6F, 0x6E, 0x74, 0x2D, 0x77, 0x65, 0x69, 0x67, 0x68, 0x74,
-	0x3A, 0x20, 0x35, 0x30, 0x70, 0x78, 0x3B, 0x0D, 0x0A, 0x20,
-	0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x63, 0x6F, 0x6C,
-	0x6F, 0x72, 0x3A, 0x23, 0x38, 0x34, 0x30, 0x30, 0x37, 0x34,
-	0x3B, 0x0D, 0x0A, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x7D,
-	0x0D, 0x0A, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x70, 0x20,
-	0x7B, 0x0D, 0x0A, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20,
-	0x20, 0x66, 0x6F, 0x6E, 0x74, 0x2D, 0x73, 0x69, 0x7A, 0x65,
-	0x3A, 0x20, 0x31, 0x2E, 0x35, 0x72, 0x65, 0x6D, 0x3B, 0x0D,
-	0x0A, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x74,
-	0x65, 0x78, 0x74, 0x2D, 0x61, 0x6C, 0x69, 0x67, 0x6E, 0x3A,
-	0x20, 0x63, 0x65, 0x6E, 0x74, 0x65, 0x72, 0x3B, 0x0D, 0x0A,
-	0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x66, 0x6F,
-	0x6E, 0x74, 0x2D, 0x77, 0x65, 0x69, 0x67, 0x68, 0x74, 0x3A,
-	0x20, 0x35, 0x30, 0x30, 0x70, 0x78, 0x3B, 0x0D, 0x0A, 0x20,
-	0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x63, 0x6F, 0x6C,
-	0x6F, 0x72, 0x3A, 0x20, 0x23, 0x66, 0x66, 0x66, 0x3B, 0x0D,
-	0x0A, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x66,
-	0x6F, 0x6E, 0x74, 0x2D, 0x77, 0x65, 0x69, 0x67, 0x68, 0x74,
-	0x3A, 0x20, 0x37, 0x30, 0x30, 0x3B, 0x0D, 0x0A, 0x20, 0x20,
-	0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x6D, 0x61, 0x72, 0x67,
-	0x69, 0x6E, 0x2D, 0x62, 0x6F, 0x74, 0x74, 0x6F, 0x6D, 0x3A,
-	0x20, 0x31, 0x72, 0x65, 0x6D, 0x3B, 0x0D, 0x0A, 0x20, 0x20,
-	0x20, 0x20, 0x20, 0x20, 0x7D, 0x0D, 0x0A, 0x20, 0x20, 0x20,
-	0x20, 0x20, 0x20, 0x61, 0x20, 0x7B, 0x0D, 0x0A, 0x20, 0x20,
-	0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x70, 0x61, 0x64, 0x64,
-	0x69, 0x6E, 0x67, 0x3A, 0x20, 0x31, 0x35, 0x70, 0x78, 0x20,
-	0x32, 0x30, 0x70, 0x78, 0x3B, 0x0D, 0x0A, 0x20, 0x20, 0x20,
-	0x20, 0x20, 0x20, 0x20, 0x20, 0x62, 0x61, 0x63, 0x6B, 0x67,
-	0x72, 0x6F, 0x75, 0x6E, 0x64, 0x3A, 0x20, 0x23, 0x35, 0x32,
-	0x63, 0x61, 0x65, 0x65, 0x3B, 0x0D, 0x0A, 0x20, 0x20, 0x20,
-	0x20, 0x20, 0x20, 0x20, 0x20, 0x66, 0x6F, 0x6E, 0x74, 0x2D,
-	0x73, 0x69, 0x7A, 0x65, 0x3A, 0x20, 0x31, 0x72, 0x65, 0x6D,
-	0x3B, 0x0D, 0x0A, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20,
-	0x20, 0x74, 0x65, 0x78, 0x74, 0x2D, 0x64, 0x65, 0x63, 0x6F,
-	0x72, 0x61, 0x74, 0x69, 0x6F, 0x6E, 0x3A, 0x20, 0x6E, 0x6F,
-	0x6E, 0x65, 0x3B, 0x0D, 0x0A, 0x20, 0x20, 0x20, 0x20, 0x20,
-	0x20, 0x20, 0x20, 0x63, 0x6F, 0x6C, 0x6F, 0x72, 0x3A, 0x20,
-	0x23, 0x33, 0x33, 0x33, 0x33, 0x33, 0x33, 0x3B, 0x0D, 0x0A,
-	0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x62, 0x6F,
-	0x72, 0x64, 0x65, 0x72, 0x2D, 0x72, 0x61, 0x64, 0x69, 0x75,
-	0x73, 0x3A, 0x20, 0x2E, 0x32, 0x35, 0x72, 0x65, 0x6D, 0x3B,
-	0x0D, 0x0A, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20,
-	0x62, 0x6F, 0x78, 0x2D, 0x73, 0x68, 0x61, 0x64, 0x6F, 0x77,
-	0x3A, 0x20, 0x30, 0x20, 0x30, 0x20, 0x32, 0x30, 0x70, 0x78,
-	0x20, 0x72, 0x67, 0x62, 0x61, 0x28, 0x32, 0x35, 0x35, 0x2C,
-	0x20, 0x32, 0x35, 0x35, 0x2C, 0x20, 0x32, 0x35, 0x35, 0x2C,
-	0x20, 0x30, 0x2E, 0x38, 0x30, 0x38, 0x29, 0x0D, 0x0A, 0x20,
-	0x20, 0x20, 0x20, 0x20, 0x20, 0x7D, 0x0D, 0x0A, 0x20, 0x20,
-	0x20, 0x20, 0x20, 0x20, 0x66, 0x6F, 0x72, 0x6D, 0x20, 0x7B,
-	0x0D, 0x0A, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20,
-	0x70, 0x61, 0x64, 0x64, 0x69, 0x6E, 0x67, 0x3A, 0x20, 0x34,
-	0x30, 0x70, 0x78, 0x20, 0x32, 0x30, 0x70, 0x78, 0x3B, 0x0D,
-	0x0A, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x74,
-	0x65, 0x78, 0x74, 0x2D, 0x61, 0x6C, 0x69, 0x67, 0x6E, 0x3A,
-	0x20, 0x63, 0x65, 0x6E, 0x74, 0x65, 0x72, 0x3B, 0x0D, 0x0A,
-	0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x7D, 0x0D, 0x0A, 0x20,
-	0x20, 0x20, 0x20, 0x20, 0x20, 0x2E, 0x67, 0x72, 0x65, 0x65,
-	0x6E, 0x20, 0x7B, 0x0D, 0x0A, 0x20, 0x20, 0x20, 0x20, 0x20,
-	0x20, 0x20, 0x20, 0x63, 0x6F, 0x6C, 0x6F, 0x72, 0x3A, 0x20,
-	0x23, 0x31, 0x33, 0x46, 0x46, 0x30, 0x30, 0x3B, 0x0D, 0x0A,
-	0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x7D, 0x0D, 0x0A, 0x20,
-	0x20, 0x20, 0x20, 0x20, 0x20, 0x2E, 0x62, 0x6C, 0x75, 0x65,
-	0x20, 0x7B, 0x0D, 0x0A, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20,
-	0x20, 0x20, 0x63, 0x6F, 0x6C, 0x6F, 0x72, 0x3A, 0x20, 0x23,
-	0x30, 0x30, 0x32, 0x45, 0x46, 0x46, 0x3B, 0x0D, 0x0A, 0x20,
-	0x20, 0x20, 0x20, 0x20, 0x20, 0x7D, 0x0D, 0x0A, 0x20, 0x20,
-	0x20, 0x20, 0x20, 0x20, 0x2E, 0x72, 0x65, 0x64, 0x20, 0x7B,
-	0x0D, 0x0A, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20,
-	0x63, 0x6F, 0x6C, 0x6F, 0x72, 0x3A, 0x20, 0x23, 0x46, 0x46,
-	0x30, 0x30, 0x30, 0x30, 0x3B, 0x0D, 0x0A, 0x20, 0x20, 0x20,
-	0x20, 0x20, 0x20, 0x7D, 0x0D, 0x0A, 0x20, 0x20, 0x3C, 0x2F,
-	0x73, 0x74, 0x79, 0x6C, 0x65, 0x3E, 0x0D, 0x0A, 0x3C, 0x2F,
-	0x68, 0x65, 0x61, 0x64, 0x3E, 0x0D, 0x0A, 0x3C, 0x62, 0x6F,
-	0x64, 0x79, 0x3E, 0x0D, 0x0A, 0x20, 0x20, 0x3C, 0x64, 0x69,
-	0x76, 0x20, 0x63, 0x6C, 0x61, 0x73, 0x73, 0x3D, 0x22, 0x77,
-	0x64, 0x22, 0x3E, 0x0D, 0x0A, 0x20, 0x20, 0x20, 0x20, 0x3C,
-	0x68, 0x31, 0x3E, 0x45, 0x6D, 0x62, 0x65, 0x54, 0x72, 0x6F,
-	0x6E, 0x69, 0x63, 0x58, 0x3C, 0x2F, 0x68, 0x31, 0x3E, 0x0D,
-	0x0A, 0x20, 0x20, 0x20, 0x20, 0x3C, 0x68, 0x34, 0x3E, 0x20,
-	0x3C, 0x69, 0x3E, 0x2D, 0x2D, 0x2D, 0x45, 0x6D, 0x62, 0x65,
-	0x64, 0x64, 0x65, 0x64, 0x20, 0x54, 0x75, 0x74, 0x6F, 0x72,
-	0x69, 0x61, 0x6C, 0x73, 0x20, 0x5A, 0x6F, 0x6E, 0x65, 0x2D,
-	0x2D, 0x2D, 0x3C, 0x2F, 0x69, 0x3E, 0x3C, 0x2F, 0x68, 0x34,
-	0x3E, 0x0D, 0x0A, 0x20, 0x20, 0x20, 0x20, 0x3C, 0x68, 0x35,
-	0x3E, 0x20, 0x53, 0x54, 0x4D, 0x33, 0x32, 0x46, 0x37, 0x36,
-	0x37, 0x5A, 0x69, 0x20, 0x45, 0x74, 0x68, 0x65, 0x72, 0x6E,
-	0x65, 0x74, 0x20, 0x48, 0x54, 0x54, 0x50, 0x20, 0x54, 0x75,
-	0x74, 0x6F, 0x72, 0x69, 0x61, 0x6C, 0x20, 0x2D, 0x20, 0x50,
-	0x61, 0x72, 0x74, 0x20, 0x32, 0x20, 0x28, 0x4C, 0x45, 0x44,
-	0x20, 0x43, 0x6F, 0x6E, 0x74, 0x72, 0x6F, 0x6C, 0x20, 0x75,
-	0x73, 0x69, 0x6E, 0x67, 0x20, 0x43, 0x47, 0x49, 0x29, 0x3C,
-	0x2F, 0x68, 0x35, 0x3E, 0x0D, 0x0A, 0x20, 0x20, 0x20, 0x20,
-	0x3C, 0x70, 0x3E, 0x3C, 0x61, 0x20, 0x68, 0x72, 0x65, 0x66,
-	0x3D, 0x22, 0x68, 0x74, 0x74, 0x70, 0x73, 0x3A, 0x2F, 0x2F,
-	0x77, 0x77, 0x77, 0x2E, 0x65, 0x6D, 0x62, 0x65, 0x74, 0x72,
-	0x6F, 0x6E, 0x69, 0x63, 0x78, 0x2E, 0x63, 0x6F, 0x6D, 0x2F,
-	0x22, 0x3E, 0x56, 0x69, 0x73, 0x69, 0x74, 0x20, 0x45, 0x6D,
-	0x62, 0x65, 0x54, 0x72, 0x6F, 0x6E, 0x69, 0x63, 0x58, 0x3C,
-	0x2F, 0x61, 0x3E, 0x3C, 0x2F, 0x70, 0x3E, 0x0D, 0x0A, 0x20,
-	0x20, 0x20, 0x20, 0x3C, 0x46, 0x4F, 0x52, 0x4D, 0x20, 0x41,
-	0x43, 0x54, 0x49, 0x4F, 0x4E, 0x3D, 0x22, 0x4C, 0x45, 0x44,
-	0x43, 0x6F, 0x6E, 0x74, 0x72, 0x6F, 0x6C, 0x2E, 0x63, 0x67,
-	0x69, 0x22, 0x20, 0x4D, 0x45, 0x54, 0x48, 0x4F, 0x44, 0x3D,
-	0x22, 0x47, 0x45, 0x54, 0x22, 0x3E, 0x0D, 0x0A, 0x20, 0x20,
-	0x20, 0x20, 0x3C, 0x64, 0x69, 0x76, 0x20, 0x63, 0x6C, 0x61,
-	0x73, 0x73, 0x3D, 0x22, 0x67, 0x72, 0x65, 0x65, 0x6E, 0x22,
-	0x3E, 0x0D, 0x0A, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x47,
-	0x72, 0x65, 0x65, 0x6E, 0x20, 0x4C, 0x45, 0x44, 0x20, 0x3C,
-	0x53, 0x45, 0x4C, 0x45, 0x43, 0x54, 0x20, 0x4E, 0x41, 0x4D,
-	0x45, 0x3D, 0x22, 0x67, 0x72, 0x65, 0x65, 0x6E, 0x22, 0x20,
-	0x53, 0x49, 0x5A, 0x45, 0x3D, 0x31, 0x3E, 0x0D, 0x0A, 0x20,
-	0x20, 0x20, 0x20, 0x20, 0x20, 0x3C, 0x4F, 0x50, 0x54, 0x49,
-	0x4F, 0x4E, 0x20, 0x53, 0x45, 0x4C, 0x45, 0x43, 0x54, 0x45,
-	0x44, 0x3E, 0x4F, 0x4E, 0x0D, 0x0A, 0x20, 0x20, 0x20, 0x20,
-	0x20, 0x20, 0x3C, 0x4F, 0x50, 0x54, 0x49, 0x4F, 0x4E, 0x3E,
-	0x4F, 0x46, 0x46, 0x0D, 0x0A, 0x20, 0x20, 0x20, 0x20, 0x20,
-	0x20, 0x3C, 0x2F, 0x53, 0x45, 0x4C, 0x45, 0x43, 0x54, 0x3E,
-	0x0D, 0x0A, 0x20, 0x20, 0x20, 0x20, 0x3C, 0x2F, 0x64, 0x69,
-	0x76, 0x3E, 0x0D, 0x0A, 0x20, 0x20, 0x20, 0x20, 0x3C, 0x64,
-	0x69, 0x76, 0x20, 0x63, 0x6C, 0x61, 0x73, 0x73, 0x3D, 0x22,
-	0x62, 0x6C, 0x75, 0x65, 0x22, 0x3E, 0x0D, 0x0A, 0x20, 0x20,
-	0x20, 0x20, 0x20, 0x20, 0x42, 0x6C, 0x75, 0x65, 0x20, 0x4C,
-	0x45, 0x44, 0x20, 0x3C, 0x53, 0x45, 0x4C, 0x45, 0x43, 0x54,
-	0x20, 0x4E, 0x41, 0x4D, 0x45, 0x3D, 0x22, 0x62, 0x6C, 0x75,
-	0x65, 0x22, 0x20, 0x53, 0x49, 0x5A, 0x45, 0x3D, 0x31, 0x3E,
-	0x0D, 0x0A, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x3C, 0x4F,
-	0x50, 0x54, 0x49, 0x4F, 0x4E, 0x20, 0x53, 0x45, 0x4C, 0x45,
-	0x43, 0x54, 0x45, 0x44, 0x3E, 0x4F, 0x4E, 0x0D, 0x0A, 0x20,
-	0x20, 0x20, 0x20, 0x20, 0x20, 0x3C, 0x4F, 0x50, 0x54, 0x49,
-	0x4F, 0x4E, 0x3E, 0x4F, 0x46, 0x46, 0x0D, 0x0A, 0x20, 0x20,
-	0x20, 0x20, 0x20, 0x20, 0x3C, 0x2F, 0x53, 0x45, 0x4C, 0x45,
-	0x43, 0x54, 0x3E, 0x0D, 0x0A, 0x20, 0x20, 0x20, 0x20, 0x3C,
-	0x2F, 0x64, 0x69, 0x76, 0x3E, 0x0D, 0x0A, 0x20, 0x20, 0x20,
-	0x20, 0x3C, 0x64, 0x69, 0x76, 0x20, 0x63, 0x6C, 0x61, 0x73,
-	0x73, 0x3D, 0x22, 0x72, 0x65, 0x64, 0x22, 0x3E, 0x0D, 0x0A,
-	0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x52, 0x65, 0x64, 0x20,
-	0x4C, 0x45, 0x44, 0x20, 0x3C, 0x53, 0x45, 0x4C, 0x45, 0x43,
-	0x54, 0x20, 0x4E, 0x41, 0x4D, 0x45, 0x3D, 0x22, 0x72, 0x65,
-	0x64, 0x22, 0x20, 0x53, 0x49, 0x5A, 0x45, 0x3D, 0x31, 0x3E,
-	0x0D, 0x0A, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x3C, 0x4F,
-	0x50, 0x54, 0x49, 0x4F, 0x4E, 0x20, 0x53, 0x45, 0x4C, 0x45,
-	0x43, 0x54, 0x45, 0x44, 0x3E, 0x4F, 0x4E, 0x0D, 0x0A, 0x20,
-	0x20, 0x20, 0x20, 0x20, 0x20, 0x3C, 0x4F, 0x50, 0x54, 0x49,
-	0x4F, 0x4E, 0x3E, 0x4F, 0x46, 0x46, 0x0D, 0x0A, 0x20, 0x20,
-	0x20, 0x20, 0x20, 0x20, 0x3C, 0x2F, 0x53, 0x45, 0x4C, 0x45,
-	0x43, 0x54, 0x3E, 0x0D, 0x0A, 0x20, 0x20, 0x20, 0x20, 0x20,
-	0x20, 0x3C, 0x62, 0x72, 0x3E, 0x0D, 0x0A, 0x20, 0x20, 0x20,
-	0x20, 0x3C, 0x2F, 0x64, 0x69, 0x76, 0x3E, 0x0D, 0x0A, 0x20,
-	0x20, 0x20, 0x20, 0x20, 0x20, 0x3C, 0x62, 0x72, 0x3E, 0x0D,
-	0x0A, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x3C, 0x69, 0x6E,
-	0x70, 0x75, 0x74, 0x20, 0x76, 0x61, 0x6C, 0x75, 0x65, 0x3D,
-	0x22, 0x53, 0x65, 0x6E, 0x64, 0x22, 0x20, 0x74, 0x79, 0x70,
-	0x65, 0x3D, 0x22, 0x73, 0x75, 0x62, 0x6D, 0x69, 0x74, 0x22,
-	0x3E, 0x0D, 0x0A, 0x20, 0x20, 0x20, 0x20, 0x3C, 0x2F, 0x64,
-	0x69, 0x76, 0x3E, 0x0D, 0x0A, 0x20, 0x20, 0x20, 0x20, 0x3C,
-	0x2F, 0x46, 0x4F, 0x52, 0x4D, 0x3E, 0x0D, 0x0A, 0x20, 0x20,
-	0x3C, 0x2F, 0x64, 0x69, 0x76, 0x3E, 0x0D, 0x0A, 0x3C, 0x2F,
-	0x62, 0x6F, 0x64, 0x79, 0x3E, 0x0D, 0x0A, 0x3C, 0x2F, 0x68,
-	0x74, 0x6D, 0x6C, 0x3E};
-
-
-
-
-
-const unsigned char Led1On_Data[] ="<HTML> \
-                                    <head><title>HTTP LED Control</title>\
-									<link rel=\"icon\" href=\"data:,\"> \
-									</head> \
-                                    <body> \
-                                      <center> \
-                                        <font style = \"color:red\">aaabbbcccddd</font> \
-                                        <form method=\"post\" action=\"/off\" name=\"ledform\"> \
-                                          <button type=\"submit\" style=\"width:80px;height:30px;\"> OFF </button> \
-                                        </form> \
-                                      </center> \
-                                    </body> \
-                                    </HTML> ";
-
-const unsigned char Led1Off_Data[] =" <HTML> \
-                                    <head>\
-									<title>HTTP LED Control</title>\
-									<link rel=\"icon\" href=\"data:,\"> \
-									</head> \
-                                    <body> \
-                                    <center> \
-                                        <font style = \"color:red\">aaabbbcccddd</font> \
-                                        <form method=\"post\" action=\"/on\" name=\"ledform\"> \
-                                        <button type=\"submit\" style=\"width:80px;height:30px;\"> ON </button> \
-                                        </form> \
-                                    </center> \
-                                    </body> \
-                                    </HTML> ";
 
 static const char http_index_html[] = "<html> \
 										<head> \
@@ -9449,13 +9004,14 @@ const unsigned char login_html[] ="<html lang=\"zh\"> \
 											align-items: center; \
 										} \
 										label { \
-											width: 200px; /* 设置label的固定宽度以确保对齐 */ \
+											width: 200px;\
 										} \
 										</style> \
                                     </head> \
-                                    <body> \
+                                    <body > \
+									<div style=\"display: flex; flex-direction: column; align-items: center;\"> \
                                         <h2>User Login</h2> \
-                                        <form action=\"/login\" method=\"POST\"> \
+                                        <form action=\"/login\" method=\"POST\"  > \
 												<div class=\"network-row\"> \
 													<label for=\"username\">Username:</label> <input type=\"text\" id=\"username\" name=\"username\" required> \
 												</div> \
@@ -9466,6 +9022,7 @@ const unsigned char login_html[] ="<html lang=\"zh\"> \
                                                 <button type=\"submit\">Login</button> \
                                             </div> \
                                         </form> \
+									</div> \
                                     </body> \
                                     </html> ";
 
@@ -9484,11 +9041,10 @@ const unsigned char info_html[] ="<html lang=\"en\"> \
                                                         var gateway = $('#gateway').val();\n \
                                                         var subnetwork = $('#subnetwork').val();\n \
 														var macaddr = $('#macaddr').val();\n \
-                                                        // 使用AJAX发送POST请求到服务器的/network路径 \n \
                                                         $.ajax({ \n\
                                                             url: '/network',\n \
                                                             type: 'POST',\n \
-                                                            contentType: 'application/x-www-form-urlencoded', // 设置Content-Type \n \
+                                                            contentType: 'application/x-www-form-urlencoded', \n \
                                                             data: {\n \
                                                                 ipaddr: ipaddr,\n \
                                                                 gateway: gateway,\n \
@@ -9501,12 +9057,10 @@ const unsigned char info_html[] ="<html lang=\"en\"> \
                                                                 }else{\n \
                                                                     alert(response.message);\n \
                                                                 }\n \
-                                                                // 请求成功时的回调函数\n \
                                                                 console.log('Network settings updated successfully.');\n \
                                                             },\n \
                                                             error: function(xhr, status, error) {\n \
                                                                 alert(\"udpate failed!\");\n \
-                                                                // 请求失败时的回调函数\n \
                                                                 console.error('Error updating network settings:', error);\n \
                                                             }\n \
                                                         });\n \
@@ -9529,7 +9083,6 @@ const unsigned char info_html[] ="<html lang=\"en\"> \
                                                                 console.log('Network settings refreshed successfully.');\n \
                                                             },\n \
                                                             error: function(xhr, status, error) {\n \
-                                                                // 请求失败时的回调函数\n \
                                                                 console.error('Error refreshing network settings:', error);\n \
                                                             }\n \
                                                         });\n\
@@ -9559,7 +9112,6 @@ const unsigned char info_html[] ="<html lang=\"en\"> \
                                                             },\n \
                                                             error: function(xhr, status, error) {\n \
                                                                 alert(\"udpate failed!\");\n \
-                                                                // 请求失败时的回调函数\n \
                                                                 console.error('Error updating power-on-change settings:', error);\n \
                                                             }\n \
                                                         });\n \
@@ -9569,7 +9121,7 @@ const unsigned char info_html[] ="<html lang=\"en\"> \
                                                         $.ajax({\n \
                                                             url: '/power_lostresume_status',\n \
                                                             type: 'POST',\n \
-                                                            contentType: 'application/x-www-form-urlencoded', // 设置Content-Type \n \
+                                                            contentType: 'application/x-www-form-urlencoded',  \n \
                                                             data: {\n \
                                                                 power_lostresume_status: power_lostresume_status,\n \
                                                             },\
@@ -9584,7 +9136,6 @@ const unsigned char info_html[] ="<html lang=\"en\"> \
                                                             },\n \
                                                             error: function(xhr, status, error) {\n \
                                                                 alert(\"udpate failed!\");\n \
-                                                                // 请求失败时的回调函数\n \
                                                                 console.error('Error updating power-lostresume-change settings:', error);\n \
                                                             }\n \
                                                         });\n \
@@ -9593,19 +9144,17 @@ const unsigned char info_html[] ="<html lang=\"en\"> \
                                                         $.ajax({ \n\
                                                             url: '/reset',\n \
                                                             type: 'POST',\n \
-                                                            contentType: 'application/x-www-form-urlencoded', // 设置Content-Type \n \
+                                                            contentType: 'application/x-www-form-urlencoded', \n \
                                                             success: function(response) {\n \
                                                                 if(response.status===0){\n \
                                                                     alert(\"reset success!\");\n \
                                                                 }else{\n \
                                                                     alert(response.message);\n \
                                                                 }\n \
-                                                                // 请求成功时的回调函数\n \
                                                                 console.log('Reset successfully.');\n \
                                                             },\n \
                                                             error: function(xhr, status, error) {\n \
                                                                 alert(\"reset failed!\");\n \
-                                                                // 请求失败时的回调函数\n \
                                                                 console.error('Error Reset:', error);\n \
                                                             }\n \
                                                         });\n \
@@ -9620,14 +9169,16 @@ const unsigned char info_html[] ="<html lang=\"en\"> \
                                                             },\
                                                             success: function(response) {\n \
 																if(response.status===0){\n \
-																	$('#power_consum').val(response.data.consumption);\n \
-																	$('#power_cur').val(response.data.current);\n \
-																	$('#power_vol').val(response.data.voltage);\n \
+																	var consumption = parseFloat(response.data.consumption)/ 1000000; \n \
+																	$('#power_consum').val(consumption.toFixed(2)); \n \
+																	var current = parseFloat(response.data.current)/ 1000; \n \
+																	$('#power_cur').val(current.toFixed(2));\n \
+																	var voltage = parseFloat(response.data.voltage)/ 1000; \n \
+																	$('#power_vol').val(voltage.toFixed(2));\n \
 																} \n \
                                                                 console.log('power_consum refreshed successfully.');\n \
                                                             },\n \
                                                             error: function(xhr, status, error) {\n \
-                                                                // 请求失败时的回调函数\n \
                                                                 console.error('Error refreshing power_consum:', error);\n \
                                                             }\n \
                                                         });\n\
@@ -9648,14 +9199,16 @@ const unsigned char info_html[] ="<html lang=\"en\"> \
 															contentType: 'application/x-www-form-urlencoded', \n \
                                                             success: function(response) {\n \
 																if(response.status===0){\n \
-																	$('#cpu_temp').val(response.data.cpu_temp);\n \
-																	$('#npu_temp').val(response.data.npu_temp);\n \
-																	$('#fan_speed').val(response.data.fan_speed);\n \
+																	var cpu_temp = parseFloat(response.data.cpu_temp)/ 1000; \n \
+																	$('#cpu_temp').val(cpu_temp.toFixed(2));\n \
+																	var npu_temp = parseFloat(response.data.npu_temp)/ 1000; \n \
+																	$('#npu_temp').val(npu_temp.toFixed(2));\n \
+																	var fan_speed = parseFloat(response.data.fan_speed) \n \
+																	$('#fan_speed').val(fan_speed);\n \
 																}\n \
                                                                 console.log('pvt info refreshed successfully.');\n \
                                                             },\n \
                                                             error: function(xhr, status, error) {\n \
-                                                                // 请求失败时的回调函数\n \
                                                                 console.error('Error refreshing pvt info:', error);\n \
                                                             }\n \
                                                         });\n \
@@ -9685,7 +9238,6 @@ const unsigned char info_html[] ="<html lang=\"en\"> \
                                                                 console.log('dip_switch refreshed successfully.');\n \
                                                             },\n \
                                                             error: function(xhr, status, error) {\n \
-                                                                // 请求失败时的回调函数\n \
                                                                 console.error('Error refreshing dip_switch:', error);\n \
                                                             }\n \
                                                         });\n\
@@ -9704,7 +9256,7 @@ const unsigned char info_html[] ="<html lang=\"en\"> \
                                                         $.ajax({ \n\
                                                             url: '/dip_switch',\n \
                                                             type: 'POST',\n \
-                                                            contentType: 'application/x-www-form-urlencoded', // 设置Content-Type \n \
+                                                            contentType: 'application/x-www-form-urlencoded', \n \
                                                             data: {\n \
                                                                 dip01: dip01, \n \
                                                                 dip02: dip02, \n \
@@ -9718,12 +9270,10 @@ const unsigned char info_html[] ="<html lang=\"en\"> \
                                                                 }else{\n \
                                                                     alert(response.message);\n \
                                                                 }\n \
-                                                                // 请求成功时的回调函数\n \
                                                                 console.log('Network settings updated successfully.');\n \
                                                             },\n \
                                                             error: function(xhr, status, error) {\n \
                                                                 alert(\"udpate failed!\");\n \
-                                                                // 请求失败时的回调函数\n \
                                                                 console.error('Error updating network settings:', error);\n \
                                                             }\n \
                                                         });\n \
@@ -9807,7 +9357,6 @@ const unsigned char info_html[] ="<html lang=\"en\"> \
 																console.log('board-info-som-refresh refreshed successfully.');\n \
                                                             },\n \
                                                             error: function(xhr, status, error) {\n \
-                                                                // 请求失败时的回调函数\n \
                                                                 console.error('Error refreshing board-info-som-refresh:', error);\n \
                                                             }\n \
                                                         });\n \
@@ -9828,7 +9377,6 @@ const unsigned char info_html[] ="<html lang=\"en\"> \
                                                                 console.log('board-info-cb-refresh refreshed successfully.');\n \
                                                             },\n \
                                                             error: function(xhr, status, error) {\n \
-                                                                // 请求失败时的回调函数\n \
                                                                 console.error('Error refreshing board-info-cb-refresh:', error);\n \
                                                             }\n \
                                                         });\n \
@@ -9861,7 +9409,6 @@ const unsigned char info_html[] ="<html lang=\"en\"> \
                                                                 console.log('rtc refreshed successfully.');\n \
                                                             },\n \
                                                             error: function(xhr, status, error) {\n \
-                                                                // 请求失败时的回调函数\n \
                                                                 console.error('Error refreshing rtc:', error);\n \
                                                             }\n \
                                                         });\n \
@@ -9882,7 +9429,7 @@ const unsigned char info_html[] ="<html lang=\"en\"> \
                                                         $.ajax({ \n\
                                                             url: '/rtc',\n \
                                                             type: 'POST',\n \
-                                                            contentType: 'application/x-www-form-urlencoded', // 设置Content-Type \n \
+                                                            contentType: 'application/x-www-form-urlencoded',  \n \
                                                             data: {\n \
                                                                 year: rtc_year, \n \
                                                                 month: rtc_month, \n \
@@ -9898,12 +9445,10 @@ const unsigned char info_html[] ="<html lang=\"en\"> \
                                                                 }else{\n \
                                                                     alert(response.message);\n \
                                                                 }\n \
-                                                                // 请求成功时的回调函数\n \
                                                                 console.log('rtc settings updated successfully.');\n \
                                                             },\n \
                                                             error: function(xhr, status, error) {\n \
                                                                 alert(\"udpate failed!\");\n \
-                                                                // 请求失败时的回调函数\n \
                                                                 console.error('Error updating rtc settings:', error);\n \
                                                             }\n \
                                                         });\n \
@@ -9927,7 +9472,6 @@ const unsigned char info_html[] ="<html lang=\"en\"> \
                                                                 console.log('rtc refreshed successfully.');\n \
                                                             },\n \
                                                             error: function(xhr, status, error) {\n \
-                                                                // 请求失败时的回调函数\n \
                                                                 console.error('Error refreshing rtc:', error);\n \
                                                             }\n \
                                                         });\n \
@@ -9977,7 +9521,7 @@ const unsigned char info_html[] ="<html lang=\"en\"> \
 												align-items: center; \
 											} \
 											label { \
-												width: 200px; /* 设置label的固定宽度以确保对齐 */ \
+												width: 200px; \
 											} \
                                             </style> \
                                     </head> \
@@ -10050,13 +9594,13 @@ const unsigned char info_html[] ="<html lang=\"en\"> \
 										<div> \
 											<h3>Power Consumption</h3> \
 											<div class=\"network-row\"> \
-												<label>Power Consumption:</label> <input type=\"text\" id=\"power_consum\" value=\"0\" style=\"width: 60px;\" disabled><span>mW</span><br> \
+												<label>Power Consumption:</label> <input type=\"text\" id=\"power_consum\" value=\"0\" style=\"width: 60px;\" disabled><span>W</span><br> \
 											</div> \
 											<div class=\"network-row\"> \
-												<label>Voltage:</label> <input type=\"text\" id=\"power_vol\" value=\"0\" style=\"width: 60px;\" disabled><span>mV</span><br> \
+												<label>Voltage:</label> <input type=\"text\" id=\"power_vol\" value=\"0\" style=\"width: 60px;\" disabled><span>V</span><br> \
 											</div> \
 											<div class=\"network-row\"> \
-												<label>Current:</label> <input type=\"text\" id=\"power_cur\" value=\"0\" style=\"width: 60px;\" disabled><span>mA</span><br> \
+												<label>Current:</label> <input type=\"text\" id=\"power_cur\" value=\"0\" style=\"width: 60px;\" disabled><span>A</span><br> \
 											</div> \
 											<button id=\"power-consum-refresh\">refresh</button> <br> \
 											<button id=\"power-consum-refresh-hid\" value=\"0\" style=\"display:none;\" >refresh</button> \
@@ -10064,13 +9608,13 @@ const unsigned char info_html[] ="<html lang=\"en\"> \
                                         <div class=\"pvt-info\" > \
                                             <h3>PVT info</h3> \
 											<div class=\"network-row\"> \
-                                            <label>CPU Temperature:</label> <input type=\"text\" id=\"cpu_temp\" value=\"0\" style=\"width: 60px;\" disabled><br> \
+                                            <label>CPU Temperature:</label> <input type=\"text\" id=\"cpu_temp\" value=\"0\" style=\"width: 60px;\" disabled><span>°C</span><br> \
 											</div> \
 											<div class=\"network-row\"> \
-                                            <label>NPU Temperature:</label> <input type=\"text\" id=\"npu_temp\" value=\"0\" style=\"width: 60px;\" disabled><br> \
+                                            <label>NPU Temperature:</label> <input type=\"text\" id=\"npu_temp\" value=\"0\" style=\"width: 60px;\" disabled><span>°C</span><br> \
 											</div> \
 											<div class=\"network-row\"> \
-                                            <label>Fan Speed      :</label> <input type=\"text\" id=\"fan_speed\" value=\"0\" style=\"width: 60px;\" disabled><br> \
+                                            <label>Fan Speed      :</label> <input type=\"text\" id=\"fan_speed\" value=\"0\" style=\"width: 60px;\" disabled><span>RPM</span><br> \
 											</div> \
                                             <button id=\"pvt-info-refresh\" >refresh</button> \
 											<button id=\"pvt-info-refresh-hid\" value=\"0\" style=\"display:none;\">refresh</button> \
@@ -10176,7 +9720,7 @@ const unsigned char modify_account_html[] ="<html lang=\"zh\"> \
 												align-items: center; \
 											} \
 											label { \
-												width: 200px; /* 设置label的固定宽度以确保对齐 */ \
+												width: 200px;  \
 											} \
                                             </style> \
                                     </head> \
@@ -10189,8 +9733,9 @@ const unsigned char modify_account_html[] ="<html lang=\"zh\"> \
 												<button type=\"submit\">Exit</button> \
 											</form> \
                                         </div> \
+										<div style=\"display: flex; flex-direction: column; align-items: center;\"> \
                                         <h2>Account modify</h2> \
-                                        <form action=\"/modify_account\" method=\"POST\"> \
+                                        <form action=\"/modify_account\" method=\"POST\"  > \
 											<div class=\"network-row\"> \
 												<label for=\"username\">Username:</label> <input type=\"text\" id=\"username\" name=\"username\" value=\"%s\" required disabled> \
 											</div> \
@@ -10201,6 +9746,7 @@ const unsigned char modify_account_html[] ="<html lang=\"zh\"> \
                                                 <button type=\"submit\">Update</button> \
                                             </div> \
                                         </form> \
+										</div> \
                                     </body> \
                                     </html> ";
 
@@ -10213,22 +9759,18 @@ const char *json_header_withcookie = "HTTP/1.1 200 OK\r\n"
 						"Connection: close\r\n"
 						"%.80s"
 						"Content-Length: %d\r\n\r\n";
-// 函数定义
+
 char* concatenate_strings(const char* str1, const char* str2) {
-    // 计算新字符串需要的总长度
     int length = strlen(str1) + strlen(str2) + 1; // +1 for the null terminator
-
-    // 动态分配内存
     char* new_str = malloc(length);
     if (new_str == NULL) {
         printf(stderr, "Memory allocation failed\n");
         exit(1);
     }
 
-    // 初始化内存区域
-    strcpy(new_str, str1);   // 复制第一个字符串
-    strcat(new_str, str2);   // 拼接第二个字符串
-    return new_str; // 返回新字符串
+    strcpy(new_str, str1);   
+    strcat(new_str, str2);  
+    return new_str; 
 }
 
 void send_redirect(struct netconn *conn, const char *location,const char* cookies) {
@@ -10236,7 +9778,6 @@ void send_redirect(struct netconn *conn, const char *location,const char* cookie
 	assert(strlen(cookies)<80);//location max length is 30,cookie maxlength 80
 	assert(strlen(location)<30);
     if(cookies!=NULL && strlen(cookies)>0){
-        // 构造一个HTTP 302 Found响应
         sprintf(header,
                 "HTTP/1.1 302 Found\r\n"
                 "Location: %.30s\r\n"
@@ -10259,24 +9800,23 @@ void send_redirect(struct netconn *conn, const char *location,const char* cookie
 
 err_t send_large_data(struct netconn *conn, const char *data, unsigned int length) {
     err_t result = ERR_OK;
-    unsigned int offset = 0; // 当前发送偏移量
+    unsigned int offset = 0; 
 
-    // 循环发送数据，直到发送完所有数据
     while (offset < length) {
         unsigned int chunk_size = CHUNK_SIZE;
         if (offset + CHUNK_SIZE > length) {
-            chunk_size = length - offset; // 最后一块可能不足CHUNK_SIZE
+            chunk_size = length - offset; 
         }
 
-        // 发送当前块
+  
         result = netconn_write(conn, data + offset, chunk_size, NETCONN_COPY);
 		printf("offset:%d chunk_size:%d \n" ,offset,chunk_size);
 
         if (result != ERR_OK) {
-            break; // 如果发送失败，则提前退出
+            break; 
         }
 
-        offset += chunk_size; // 更新偏移量
+        offset += chunk_size; 
     }
 
     return result;
@@ -10321,20 +9861,17 @@ void send_response_200(struct netconn *conn) {
 
 // ------------------------ parse params  ---------------------
 
-// 定义一个结构体来代表键值对
 typedef struct kv_pair {
     char *key;
     char *value;
     struct kv_pair *next;
 } kv_pair;
 
-// 定义一个结构体来代表一个键值对的链表
 typedef struct {
     kv_pair *head;
     size_t size;
 } kv_map;
 
-// 创建一个新的键值对节点
 kv_pair* create_kv_pair(const char *key, const char *value) {
     kv_pair *new_pair = (kv_pair *)malloc(sizeof(kv_pair));
     if (new_pair == NULL) {
@@ -10347,7 +9884,6 @@ kv_pair* create_kv_pair(const char *key, const char *value) {
     return new_pair;
 }
 
-// 清空并释放链表
 void free_kv_map(kv_map *map) {
     kv_pair *current = map->head;
     while (current != NULL) {
@@ -10361,21 +9897,17 @@ void free_kv_map(kv_map *map) {
     map->size = 0;
 }
 
-
-
-// 解析GET请求参数列表
 kv_map parse_get_params(const char *query) {
     kv_map params = {NULL, 0};
-    char *query_dup = strdup(query); // 复制查询字符串以便修改
+    char *query_dup = strdup(query); 
     char *token = strtok(query_dup, "&");
 
     while (token) {
         char *equal_sign = strchr(token, '=');
         if (equal_sign) {
-            *equal_sign = '\0'; // 分隔键和值
+            *equal_sign = '\0'; 
             kv_pair *new_pair = create_kv_pair(token, equal_sign + 1);
             if (new_pair) {
-                // 将新节点插入链表头部（简单起见，不维护有序）
                 new_pair->next = params.head;
                 params.head = new_pair;
                 params.size++;
@@ -10412,35 +9944,30 @@ void parse_cookies(const char* request, char* cookies) {
 
 
 
-// 函数用于生成一个随机字符串作为会话ID
 void generate_session_id(char *session_id, int length) {
-    // 定义可能出现在会话ID中的字符
     const char charset[] = "0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ";
     if (length > 0) {
-        // 设置种子
         srand((unsigned int)time(NULL));
-        // 生成会话ID
         for (int i = 0; i < length - 1; ++i) {
             int key = rand() % (int)(sizeof(charset) - 1);
             session_id[i] = charset[key];
         }
-        session_id[length - 1] = '\0';  // 确保最后是空字符以结束字符串
+        session_id[length - 1] = '\0';  
     }
 }
 
 
-// 会话结构体
 typedef struct Session {
-    char session_id[256];  // 会话ID
-    char session_data[256]; // 会话数据
-    struct Session *next;  // 指向下一个会话的指针
+    char session_id[256];  
+    char session_data[256]; 
+    struct Session *next; 
 } Session;
 
 
-// 会话链表头
+
 Session *session_list = NULL;
 
-// 创建一个新的会话
+
 Session* create_session(const char *id, const char *data) {
     Session *new_session = (Session *)malloc(sizeof(Session));
     if (new_session) {
@@ -10453,13 +9980,11 @@ Session* create_session(const char *id, const char *data) {
     return new_session;
 }
 
-// 添加会话到会话列表
 void add_session(Session *session) {
     session->next = session_list;
     session_list = session;
 }
 
-// 查找会话
 Session* find_session(const char *id) {
     Session *current = session_list;
     while (current != NULL) {
@@ -10471,18 +9996,16 @@ Session* find_session(const char *id) {
     return NULL;
 }
 
-// 更新会话
 int update_session(const char *id, const char *data) {
     Session *session = find_session(id);
     if (session) {
         strncpy(session->session_data, data, sizeof(session->session_data) - 1);
         session->session_data[sizeof(session->session_data) - 1] = '\0';
-        return 1; // 更新成功
+        return 1;
     }
-    return 0; // 更新失败
+    return 0; 
 }
 
-// 删除会话
 int delete_session(const char *id) {
     Session **current = &session_list;
     while (*current != NULL) {
@@ -10490,14 +10013,13 @@ int delete_session(const char *id) {
             Session *to_delete = *current;
             *current = (*current)->next;
             free(to_delete);
-            return 1; // 删除成功
+            return 1; 
         }
         current = &(*current)->next;
     }
-    return 0; // 删除失败
+    return 0; 
 }
 
-// 清理所有会话
 void clear_sessions() {
     Session *current = session_list;
     while (current != NULL) {
@@ -10514,27 +10036,26 @@ void clear_sessions() {
 
 void parseCredentials(const char *readBuffer, char *username, char *password) {
     char buffer[EEPROM_USERNAME_PASSWORD_BUFFER_SIZE];
-    strncpy(buffer, readBuffer, EEPROM_USERNAME_PASSWORD_BUFFER_SIZE); // 拷贝到临时变量以避免更改原字符串
-    buffer[EEPROM_USERNAME_PASSWORD_BUFFER_SIZE - 1] = '\0'; // 确保字符串以空字符结尾
+    strncpy(buffer, readBuffer, EEPROM_USERNAME_PASSWORD_BUFFER_SIZE); 
+    buffer[EEPROM_USERNAME_PASSWORD_BUFFER_SIZE - 1] = '\0'; 
 
-    // 使用strtok分割字符串
     char *token = strtok(buffer, ",");
     if (token != NULL) {
-        strncpy(username, token, EEPROM_USERNAME_PASSWORD_BUFFER_SIZE); // 拷贝用户名
+        strncpy(username, token, EEPROM_USERNAME_PASSWORD_BUFFER_SIZE); 
         token = strtok(NULL, ",");
         if (token != NULL) {
-            strncpy(password, token, EEPROM_USERNAME_PASSWORD_BUFFER_SIZE); // 拷贝密码
+            strncpy(password, token, EEPROM_USERNAME_PASSWORD_BUFFER_SIZE);
         }
     }
 }
 
-//保存用户设置的用户名密码 0 succ,1 fail
+// 0 succ,1 fail
 int save_sys_username_password(const char *username, const char *password){
 	assert(username!=NULL&&password!=NULL);
 
 	return es_set_username_password(username, password);
 }
-//查询用户设置的用户名密码 0 succ,1 fail
+// 0 succ,1 fail
 void get_sys_username_password( char *username,  char *password){
 
 	es_get_username_password(username, password);
@@ -10561,56 +10082,33 @@ const char* process_header(const char *header) {
     char line[1024];
 
     while ((end = strstr(pos, "\r\n")) != NULL) {
-        // 复制一行到临时缓冲区，注意要避免缓冲区溢出
         size_t len = end - pos;
         if (len > sizeof(line) - 1) len = sizeof(line) - 1;
         memcpy(line, pos, len);
-        line[len] = '\0'; // 确保字符串以null字符终止
-
-        // 处理这一行
+        line[len] = '\0'; 
         printf("Header line :pos:%d strlen:%d line:%c%c%c%c%c\n",pos-header,strlen(line), line[0],line[1],line[2],line[3],line[4]);
 
-        pos = end + 2; // 跳过"\r\n"
+        pos = end + 2; // skip "\r\n"
 
-        // 检测头部结束（即"\r\n\r\n"）
+        // check header（即"\r\n\r\n"）
         if (*pos == '\r' && *(pos + 1) == '\n') {
             printf("End of header.\n");
-            return pos + 2; // 返回指向响应体开始的指针
+            return pos + 2;
         }
     }
 	printf("can't find rnrn pos-header:%d .\n",pos-header);
-    return NULL; // 如果没有找到"\r\n\r\n"，返回NULL
+    return NULL; // not found "\r\n\r\n"，return NULL
 }
 
-// void send_response(int client_fd, const char* cookies) {
-//     char response[BUFFER_SIZE];
-//     char new_cookie[256];
-
-//     if (strlen(cookies) == 0) {
-//         strcpy(new_cookie, "set-cookie: user=guest; Max-Age=3600; Path=/\r\n");
-//     } else {
-//         strcpy(new_cookie, "");
-//     }
-
-//     snprintf(response, sizeof(response),
-//              "HTTP/1.1 200 OK\r\n"
-//              "Content-Type: text/html; charset=UTF-8\r\n"
-//              "%s\r\n"
-//              "Content-Length: 13\r\n"
-//              "\r\n"
-//              "Hello, World!\n",
-//              new_cookie);
-
-//     write(client_fd, response, strlenpower_status(response));
-// }
+
 static bool led_on = FALSE;
 
 
 void unescape_colon(char *str) {
     char *pos;
-    while ((pos = strstr(str, "%3A")) != NULL) {  // 查找转义序列 "%3A"
-        *pos = ':';  // 替换为冒号
-        memmove(pos + 1, pos + 3, strlen(pos + 3) + 1);  // 移动剩余的字符串
+    while ((pos = strstr(str, "%3A")) != NULL) {  
+        *pos = ':';  
+        memmove(pos + 1, pos + 3, strlen(pos + 3) + 1);  
     }
 }
 
@@ -10679,9 +10177,9 @@ int reset()
 
 
 typedef struct  {
-    int consumption;   	//功耗
-    int current;   		//电流
-    int voltage;		//电压
+    int consumption;   	
+    int current;   		
+    int voltage;		
 } POWERInfo;
 
 POWERInfo get_power_info(){
@@ -10696,19 +10194,12 @@ POWERInfo get_power_info(){
     return example;
 }
 
-int set_power_info(POWERInfo powerInfo){
-
-	printf("TODO call set_power_info\n");
-
-
-    return 0;
-}
 
 typedef struct  {
-    char ipaddr[16];    // IP地址，假设为IPv4，足够容纳xxx.xxx.xxx.xxx
-    char macaddr[18];   // MAC地址，足够容纳xx:xx:xx:xx:xx:xx
-    char subnetwork[16]; // 子网掩码
-    char gateway[16];    // 网关地址
+    char ipaddr[16];    // IPxxx.xxx.xxx.xxx
+    char macaddr[18];   // MACxx:xx:xx:xx:xx:xx
+    char subnetwork[16];
+    char gateway[16];    
 } NETInfo;
 
 
@@ -10928,12 +10419,6 @@ int get_soc_status()
      err_t err;
      bool isLogin=FALSE;
 
-    // parse file account.txt
-    // parse_accounts_to_map("account.txt", &account_list);
-    // free_kv_map(&account_list);
-
-
-     /* 读取数据 */
      err = netconn_recv(conn, &inbuf);
      printf("http_server_netconn_serve after:netconn_recv \n");
      if (err == ERR_OK)
@@ -10944,12 +10429,12 @@ int get_soc_status()
         // for (int i = 0; i < sizeof(buf)>10?10:sizeof(buf); i++) {
         //     printf("%c", buf[i]);
         // }
-        // printf("\n"); // 打印换行符
+        // printf("\n"); 
 
         char *buf_copy;
         buf_copy = strdup(buf);
 
-        // 解析GET请求,cookies
+        // parse get ,cookies
         char *method = strtok(buf_copy, " ");
         char *url = strtok(NULL, " ");
         char *version = strtok(NULL, "\r\n");
@@ -10958,7 +10443,7 @@ int get_soc_status()
         parse_cookies(buf, cookies);
         printf("parse header cookies: %s \n",cookies);
 
-        // 从cookie中获取session信息,session sid
+        // parse cookie ,session sid
         char *found_session_user_name=NULL;
         char *sidValue = NULL;
         if (cookies != NULL) {
@@ -10972,21 +10457,16 @@ int get_soc_status()
 
 
             char *cookies_copy=strdup(cookies);
-            // 使用分号和空格作为分隔符逐个检查Cookie
             char *token = strtok(cookies_copy, "; ");
             while (token != NULL) {
-                // 检查当前片段是否包含sid字段
                 if (strncmp(token, "sid=",4) == 0) {
-                    // 获取sid值，跳过"sid="这四个字符
                     sidValue = token + 4;
                     break;
                 }
-                // 继续下一个片段
                 token = strtok(NULL, "; ");
             }
 
             if (sidValue != NULL) {
-                // 找到sid值
                 printf("sidValue : %s\n", sidValue);
                 Session *found_session = find_session(sidValue);
                 if (found_session) {
@@ -11002,12 +10482,11 @@ int get_soc_status()
 
         if (method && url && version && strcmp(method, "GET") == 0) {
             printf("GET \n");
-            // 提取请求路径和查询字符串
             char *path = strtok(url, "?");
             char *query = strtok(NULL, "?");
 
             printf("GET path:%s \n",path);
-			int byhand=0;//todoooooooooooooooooooooooooooo
+			int byhand=0;
 
 
 
@@ -11020,7 +10499,6 @@ int get_soc_status()
                 params = parse_get_params(query);
                 p_params= &params;
 
-                // 打印参数列表
                 kv_pair *current = params.head;
                 while (current) {
                     printf("\t %s = %s\n", current->key, current->value);
@@ -11030,21 +10508,12 @@ int get_soc_status()
                     current = current->next;
                 }
             }else{
-                //  printf("333333333 \n");
                 printf("\t query is empty ;\n");
-                // send_response_200(conn);
-                // return ;
             }
  			if(strcmp(path, "/")==0  || strcmp(path, "/index.html")==0){
                 printf("GET location: index.html \n");
-                // strcpy(resp_cookies,"");
-                // send_200_response(conn);
-
-                //判断用户是否登录
-                //未登录->login.html
-                //已登录->info.html
 
-                if(found_session_user_name!=NULL && strlen(found_session_user_name)>0){//todo verify user_name in txt config(valid)
+                if(found_session_user_name!=NULL && strlen(found_session_user_name)>0){
                     printf("redirect to info.html (%s)\n",found_session_user_name);
                     send_redirect(conn,"/info.html",NULL);
 
@@ -11084,6 +10553,7 @@ int get_soc_status()
 					assert(strlen(found_session_user_name)<BUF_SIZE_64);
 					char modify_account_html_resp[strlen(modify_account_html)+BUF_SIZE_64];
 					sprintf(modify_account_html_resp, modify_account_html, found_session_user_name);
+					// printf("modify_account_html_resp sizeof: %d ,strlen:%d \n",sizeof(modify_account_html_resp),strlen(modify_account_html_resp));
 					sprintf(resp_cookies, "Set-Cookie: sid=%.31s; Max-Age=%d; Path=/\r\n",sidValue,MAX_AGE);
                		send_response_content(conn,resp_cookies, modify_account_html_resp);
 				}else{
@@ -11098,7 +10568,6 @@ int get_soc_status()
 				printf("GET location: power_status \n");
 				int power_status=get_power_status();
 				char json_response[BUF_SIZE_128]={0};
-				 // 创建JSON格式的字符串
                 char *json_response_patt = "{\"status\":0,\"message\":\"success\",\"data\":{\"power_status\":\"%d\"}}";//0 success, msg
                 sprintf(json_response, json_response_patt,power_status);
 
@@ -11118,7 +10587,6 @@ int get_soc_status()
 				printf("GET location: power_lostresume_status \n");
 				int power_lostresume_status=get_power_lost_resume_attr();
 				char json_response[BUF_SIZE_128]={0};
-				 // 创建JSON格式的字符串
                 char *json_response_patt = "{\"status\":0,\"message\":\"success\",\"data\":{\"power_lostresume_status\":\"%d\"}}";//0 success, msg
                 sprintf(json_response, json_response_patt,power_lostresume_status);
 
@@ -11137,7 +10605,6 @@ int get_soc_status()
 			}else if(strcmp(path, "/power_consum")==0 ){
 				POWERInfo power_info=get_power_info();
 				char json_response[BUF_SIZE_256]={0};
-				 // 创建JSON格式的字符串
                 char *json_response_patt = "{\"status\":0,\"message\":\"success\",\"data\":{\"consumption\":\"%d\",\"voltage\":\"%d\",\"current\":\"%d\"}}";//0 success, msg
                 sprintf(json_response, json_response_patt,power_info.consumption,power_info.voltage,power_info.current);
 
@@ -11169,7 +10636,6 @@ int get_soc_status()
 				int rett=get_pvt_info(ppvt_info);
 
 				char json_response[BUF_SIZE_128]={0};
-				 // 创建JSON格式的字符串
                 char *json_response_patt = "{\"status\":%d,\"message\":\"success\",\"data\":{\"cpu_temp\":\"%d\",\"npu_temp\":\"%d\",\"fan_speed\":\"%d\"}}";
                 sprintf(json_response, json_response_patt,rett,ppvt_info->cpu_temp,ppvt_info->npu_temp,ppvt_info->fan_speed);
 
@@ -11192,7 +10658,6 @@ int get_soc_status()
 				DIPSwitchInfo dipSwitchInfo= {0};
 				get_dip_switch(&dipSwitchInfo);
 				char json_response[BUF_SIZE_128]={0};
-				 // 创建JSON格式的字符串
                 char *json_response_patt = "{\"status\":0,\"message\":\"success\",\"data\":{\"dip01\":\"%d\",\"dip02\":\"%d\",\"dip03\":\"%d\",\"dip04\":\"%d\",\"swctrl\":\"%d\"}}";
                 sprintf(json_response, json_response_patt,dipSwitchInfo.dip01,dipSwitchInfo.dip02,dipSwitchInfo.dip03,dipSwitchInfo.dip04,dipSwitchInfo.swctrl);
 
@@ -11214,8 +10679,6 @@ int get_soc_status()
 
 			}else if(strcmp(path, "/network")==0 ){
 				printf("GET location: network \n");
-
-                // 创建JSON格式的字符串
 				char json_response[BUF_SIZE_256]={0};
 				NETInfo netinfo=get_net_info();
                 char *json_response_patt = "{\"status\":0,\"message\":\"success\",\"data\":{\"ipaddr\":\"%s\",\"gateway\":\"%s\",\"subnetwork\":\"%s\",\"macaddr\":\"%s\"}}";//0 success, msg
@@ -11237,7 +10700,6 @@ int get_soc_status()
                 netconn_write(conn, response_header, strlen(response_header), NETCONN_COPY);
                 netconn_write(conn, json_response, strlen(json_response), NETCONN_COPY);
 
-                // 释放JSON字符串的内存
                 // free(response_header);
                 // free(json_response);
             }else if(strcmp(path, "/fake_add_session")==0 ){
@@ -11271,7 +10733,6 @@ int get_soc_status()
 					"Content-Length: %d\r\n"\
                     "Connection: close\r\n\r\n"  ;
 
-
                 char response_header[BUF_SIZE_256];
                 sprintf(response_header, http_jquery_200_patt,  sizeof(data_jquery_min_js));
 
@@ -11297,7 +10758,6 @@ int get_soc_status()
 				int rett = get_som_info(pSimpleInfo);
 
 				char json_response[BUF_SIZE_256]={0};
-				 // 创建JSON格式的字符串
                 char *json_response_patt = "{\"status\":%d,\"message\":\"success\",\"data\":{\"magicNumber\":\"%d\",\"formatVersionNumber\":\"%d\",\"productIdentifier\":\"%d\",\"pcbRevision\":\"%d\",\"boardSerialNumber\":\"%s\"}}";
                 sprintf(json_response, json_response_patt,rett,pSimpleInfo->magic,pSimpleInfo->version,pSimpleInfo->id,pSimpleInfo->pcb,pSimpleInfo->sn);
 
@@ -11319,12 +10779,9 @@ int get_soc_status()
 				CBSimpleInfo simpleInfo=get_cb_info();
 
                	char json_response[BUF_SIZE_256]={0};
-				 // 创建JSON格式的字符串
                 char *json_response_patt = "{\"status\":0,\"message\":\"success\",\"data\":{\"magicNumber\":\"%d\",\"formatVersionNumber\":\"%d\",\"productIdentifier\":\"%d\",\"pcbRevision\":\"%d\",\"boardSerialNumber\":\"%s\"}}";
                 sprintf(json_response, json_response_patt,simpleInfo.magicNumber,simpleInfo.formatVersionNumber,simpleInfo.productIdentifier,simpleInfo.pcbRevision,simpleInfo.boardSerialNumber);
 
-
-
                 char response_header[BUF_SIZE_256];
                 if(found_session_user_name!=NULL && strlen(found_session_user_name)>0 && byhand ){
 					sprintf(resp_cookies, "Set-Cookie: sid=%.31s; Max-Age=%d; Path=/\r\n",sidValue,MAX_AGE);
@@ -11368,7 +10825,6 @@ int get_soc_status()
 				int ret=get_soc_status();
 
                	char json_response[BUF_SIZE_128]={0};
-				 // 创建JSON格式的字符串
                 char *json_response_patt = "{\"status\":0,\"message\":\"success\",\"data\":{\"status\":\"%d\"}}";
                 sprintf(json_response, json_response_patt,ret);
 
@@ -11392,8 +10848,6 @@ int get_soc_status()
                 send_response_200(conn);
             }
 
-
-            // 清空并释放get参数列表
             if(p_params!=NULL){
                 free_kv_map(&params);
                 p_params=NULL;
@@ -11414,7 +10868,7 @@ int get_soc_status()
             }else{
                 char *path = strtok(url, "?");
                 // ---------------- query info ---------------
-                // 获取 Content-Length
+                //  Content-Length
                 char *content_length_str = strstr(buf, "Content-Length: ");
                 int content_length = 0;
                 if (content_length_str) {
@@ -11429,7 +10883,6 @@ int get_soc_status()
                 printf("content_length:%d \n",content_length);
 
                 printf("########### buf hex##############\n");
-                // 以十六进制格式打印字符串
                 for (int i = 0; i < strlen(buf); i++) {
                     printf("%02X ", buf[i]);
                     if(i%16==15){
@@ -11438,15 +10891,11 @@ int get_soc_status()
                 }
                 printf("\n");
 
-
-
-                // 查找请求体的起始位置（空行后面）
                 char *body_start = strstr(buf, "\r\n\r\n");
                 // const char *body_start = process_header(buf);
 
-
                 if (body_start) {
-                    body_start += 4; // 跳过空行，指向请求体的起始位置
+                    body_start += 4; 
                     printf("find the body of the POST request. \n body_start posabs:%d strlen():%d \n",body_start-buf,strlen(body_start));
                 } else {
                     // printf("Could not find the body of the POST request. \n");
@@ -11456,7 +10905,6 @@ int get_soc_status()
 
                 char* query=malloc(sizeof(char)*(content_length+1));
 
-                // 使用 strncpy 函数截取子字符串
                 strncpy(query, body_start , content_length);
                 query[content_length] = '\0';
 
@@ -11464,7 +10912,6 @@ int get_soc_status()
                 // ------- end query info ---------------
 
                printf("########### query hex##############\n");
-    			// 以十六进制格式打印字符串
     			for (int i = 0; i < strlen(query); i++) {
     				printf("%02X,%c ", query[i],query[i]);
     				if(i%16==15){
@@ -11480,7 +10927,6 @@ int get_soc_status()
                      params = parse_get_params(query);
                      p_params = &params;
 
-                    // 打印参数列表
                     kv_pair *current = params.head;
                     while (current) {
                         printf("%s = %s\n", current->key, current->value);
@@ -11490,9 +10936,6 @@ int get_soc_status()
                 printf("end parse post query \n");
 				free(query);
 
-
-                // printf("############ path: %s compare  %d ########### \n",path,strcmp(path, "/testpost"));
-
                 if(strcmp(path, "/testpost")==0 ){		//testpost
                     printf("POST ,location: testpost \n");
 
@@ -11508,12 +10951,6 @@ int get_soc_status()
                     }
                     printf("param a: %s \n",a);
 
-
-                    led_on = TRUE;
-                    LED1_ON;
-
-                    printf("LED ON!\n");
-
                 }else if(strcmp(path, "/login")==0 ){		//login
                     printf("POST ,location: login \n");
                     char* username=NULL;
@@ -11542,8 +10979,6 @@ int get_soc_status()
 
 
 
-                    led_on = TRUE;
-                    LED1_ON;
 
                     bool loginSuccess = validate_credentials(username,password)==0?TRUE:FALSE;
 
@@ -11633,8 +11068,6 @@ int get_soc_status()
 						change_status_ret=change_power_status(1);
 					}
 
-
-					// 创建JSON格式的字符串
                     char *json_response_patt =NULL;
 					char json_response[BUF_SIZE_64]={0};
 
@@ -11682,8 +11115,6 @@ int get_soc_status()
 						change_status_ret=change_power_lost_resume_attr(1);
 					}
 
-
-					// 创建JSON格式的字符串
                     char *json_response_patt =NULL;
 					char json_response[BUF_SIZE_64]={0};
 
@@ -11711,7 +11142,6 @@ int get_soc_status()
 				}else if(strcmp(path, "/reset")==0 ){
 					printf("POST ,location: reset \n");
 					int reset_ret=reset();
-					// 创建JSON格式的字符串
                     char *json_response_patt =NULL;
 					char json_response[BUF_SIZE_64]={0};
 					if(reset_ret==0){
@@ -11766,7 +11196,6 @@ int get_soc_status()
                     }
 					int set_ret=set_dip_switch(dipSwitchInfo);
 
-					// 创建JSON格式的字符串
                     char *json_response_patt =NULL;
 					char json_response[BUF_SIZE_64]={0};
 					if(set_ret==0){
@@ -11815,7 +11244,7 @@ int get_soc_status()
                     assert(ipaddr!=NULL&&gateway!=NULL&&subnetwork!=NULL&&macaddr!=NULL);
 
 					// printf("###############  macaddr:%d %s \n",strlen(macaddr),macaddr);
-					unescape_colon(macaddr);//转义字符，冒号:
+					unescape_colon(macaddr);
 					// printf("############### macaddr:%d %s \n",strlen(macaddr),macaddr);
 
                     printf("param ipaddr: %s \n",ipaddr);
@@ -11832,7 +11261,6 @@ int get_soc_status()
 					strncpy(netinfo.subnetwork, subnetwork, strlen(subnetwork));
 					strncpy(netinfo.gateway, gateway, strlen(gateway));
 
-					// 确保字符串以空字符'\0'结尾
 					netinfo.ipaddr[strlen(ipaddr)] = '\0';
 					netinfo.macaddr[strlen(macaddr)] = '\0';
 					netinfo.subnetwork[strlen(subnetwork)] = '\0';
@@ -11840,8 +11268,6 @@ int get_soc_status()
 
 					set_net_info(netinfo);
 
-
-                     // 创建JSON格式的字符串
                     char *json_response = "{\"status\":0,\"message\":\"success\",\"data\":{}}";//0 success, msg
 
                     char response_header[BUF_SIZE_256];
@@ -11857,7 +11283,6 @@ int get_soc_status()
                     netconn_write(conn, response_header, strlen(response_header), NETCONN_COPY);
                     netconn_write(conn, json_response, strlen(json_response), NETCONN_COPY);
 
-                    // // 释放JSON字符串的内存
                     // free(response_header);
                     // free(json_response);
 
@@ -11900,7 +11325,6 @@ int get_soc_status()
 					
 					int set_ret=set_rtcinfo(rtcInfo);
 
-					// 创建JSON格式的字符串
                     char *json_response_patt =NULL;
 					char json_response[BUF_SIZE_64]={0};
 					if(set_ret==0){
@@ -11965,7 +11389,6 @@ int get_soc_status()
         printf("http_server_netconn_serve after else 33333333333333\n");
      }
 
-     /* 释放inbuf */
      netbuf_delete(inbuf);
      printf("http_server_netconn_serve after:netbuf_delete \n");
  }
@@ -11977,14 +11400,11 @@ int get_soc_status()
      struct netconn *conn, *newconn;
      err_t err;
      LWIP_UNUSED_ARG(arg);
-	/* 创建netconn连接结构 */
-	/* 绑定端口号与IP地址，端口号默认是80 */
 	conn = netconn_new(NETCONN_TCP);
 
 	LWIP_ERROR("http_server: invalid conn", (conn != NULL), return;);
 	netconn_bind(conn, NULL, LOCAL_PORT);
 
-	/* 监听 */
 	netconn_listen(conn);
 
 	led_on = TRUE;
@@ -11992,15 +11412,12 @@ int get_soc_status()
 
 	do
 	{
-		//处理连接请求/jquery.min.js
 		err = netconn_accept(conn, &newconn);
 		if (err == ERR_OK)
 		{
-			//发送网页数据
 			http_server_netconn_serve(newconn);
 
-			//删除连接结构
-			netconn_close(newconn); /* 关闭连接 */
+			netconn_close(newconn); 
 			netconn_delete(newconn);
 		}else{
 			printf("http_server_netconn_thread:err %d \n",err);
@@ -12009,7 +11426,6 @@ int get_soc_status()
 		}
 	}
 	while (1);
-	//关闭
 	netconn_close(conn);
 	netconn_delete(conn);
 	printf("http_server_netconn_thread after netconn_delete \n");
-- 
2.25.1

